osu.Game/Screens/Play/Player.cs

```diff
@@ -89,6 +89,13 @@ private void load(AudioManager audio, APIAccess api, OsuConfigManager config)
             userAudioOffset = config.GetBindable<double>(OsuSetting.AudioOffset);
 
             WorkingBeatmap working = Beatmap.Value;
+
+            if (working is DummyWorkingBeatmap)
+            {
+                Exit();
+                return;
+            }
+

```

Comment:
This should probably be the first thing done in this method, after `this.api = api;`, just so that we don't cause any unnecessary sample retrievals/etc.